sdk/python/lib/pulumi/runtime/invoke.py

```diff
@@ -72,17 +72,17 @@ def _sync_await(awaitable: Awaitable[Any]) -> Any:
     #
     # See https://github.com/python/cpython/blob/3.6/Lib/asyncio/base_events.py#L1428-L1452 for the details of the
     # _run_once kernel with which we need to cooperate.
-    ntodo = len(loop._ready)
+    ntodo = len(loop._ready) # type: ignore
     while not fut.done() and not fut.cancelled():
-        loop._run_once()
-        if loop._stopping:
+        loop._run_once() # type: ignore
+        if loop._stopping: # type: ignore
             break
     # If we drained the ready list past what a calling _run_once would have expected, fix things up by pushing
     # cancelled handles onto the list.
-    while len(loop._ready) < ntodo:
-        handle = asyncio.Handle(None, None, loop)
+    while len(loop._ready) < ntodo: # type: ignore
+        handle = asyncio.Handle(lambda: None, [], loop)
         handle._cancelled = True
-        loop._ready.append(handle)
+        loop._ready.append(handle) # type: ignore
```

Comment:
i think this is fine for now... but this does seem like a place where maybe we would like types in the future since thisis black magic and anything that can catch bugs would be nice :)  maybe file issuefor future?

Reply:
https://github.com/pulumi/pulumi/issues/3767